assets/js/beacon_admin.js

```diff
@@ -19,6 +19,43 @@ window.addEventListener("lme:editor_mounted", (ev) => {
   })
 })
 
+window.addEventListener("beacon_admin:clipcopy", (event) => {
+  const result_id = `${event.target.id}-copy-to-clipboard-result`
+  const el = document.getElementById(result_id);
+
+  if ("clipboard" in navigator) {
+    if (event.target.tagName === "INPUT") {
+      txt = event.target.value;
+    } else {
+      txt = event.target.textContent;
+    }
+
+    navigator.clipboard.writeText(txt).then(() => {
+      el.innerText = 'Copied to clipboard';
+      // Make it visible
+      el.classList.remove('invisible', 'text-red-500', 'opacity-0');
+      // Fade in and translate upwards
+      el.classList.add('text-green-500', 'opacity-100', '-translate-y-2');
+
+      setTimeout(function () {
+        el.classList.remove('text-green-500', 'opacity-100', '-translate-y-2');
+        el.classList.add('invisible', 'text-red-500', 'opacity-0');
+      }, 2000);
+
+    }).catch(() => {
+      el.innerText = 'Could not copy';
+      // Make it visible
+      el.classList.remove('invisible', 'text-green-500', 'opacity-0');
+      // Fade in and translate upwards
+      el.classList.add('text-red-500', 'opacity-100', '-translate-y-2');
+    })
+  } else {
+    alert(
+      "Sorry, your browser does not support clipboard copy.\nThis generally requires a secure origin \u2014 either HTTPS or localhost."
```

Comment:
Can we make this message more user friendly?

Reply:
done